src/sagas.js

```diff
@@ -581,6 +582,16 @@ function* fetchUSBFilter (action) {
   }
 }
 
+function* checkVmAvailability (action) {
```

Comment:
*copy from previous pull request version*

Maybe this can be simplified:

```js
yield fetchSingleVm({ payload: { vmId: action.payload.vmId } })
const vm = yield select(state => state.getIn(['vms', 'vms', action.payload.vmId])
if (!vm) {
  yield put(redirectRoute({ route: '/' }))
}
```

It could save one request.

Reply:
No, because if we call `fetchSingleVm` it update state `vms`, before adding fetched VM to state, thus `VmDetailPage` will update and call check again, and again, and again... Thus my version is more simplified. Otherwise it need to add more lines for take care of this recursion.

Reply:
I see. So how about (pseudo-code):

```js
const vmFromStore = yield select(state => state.getIn(['vms', 'vms', action.payload.vmId])
if (vmFromStore) {
  return
}
yield fetchSingleVm({ payload: { vmId: action.payload.vmId } })
const loadedVm = yield select(state => state.getIn(['vms', 'vms', action.payload.vmId])
if (!loadedVm) {
  yield put(redirectRoute({ route: '/' }))
}
```

I agree that originally posted version is easy to read. I find network requests expensive and requesting the same entity twice in this case not necessary.

Reply:
It doing the same thing, and we already check availability of VM in current `vms` state.